service/evidence/evidence_store.go

```diff
@@ -40,28 +44,43 @@ func (s *Service) StoreEvidence(_ context.Context, req *evidence.StoreEvidenceRe
 
 	_, err = e.Validate()
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "invalid req: %v", err)
+		return resp, status.Errorf(codes.InvalidArgument, "invalid req: %v", err)
 	}
 
 	s.evidences[e.Id] = e
 	resp.Status = true
+
+	// Inform our hook, if we have any
+	if s.EvidenceHook != nil {
+		for _, hook := range s.EvidenceHook {
+			go hook(e, nil)
+		}
+	}
+
 	return resp, err
 }
 
 // StoreEvidences is a method implementation of the evidenceServer interface: It receives evidences and stores them
 func (s *Service) StoreEvidences(stream evidence.EvidenceStore_StoreEvidencesServer) (err error) {
 	var (
-		req *evidence.StoreEvidenceRequest
-		e   *evidence.Evidence
+		req              *evidence.StoreEvidenceRequest
+		receivedEvidence *evidence.Evidence
```

Comment:
I like the shorter representation `e` more because it has a short scope. I like the convention of the length of variable names being proportional to the scope.